src/lib/Parser.hs

```diff
@@ -271,12 +271,11 @@ interfaceDef = do
   (oneAnnBinder, pos) <- withPos annBinder
   keyWord WhereKW
   record <- withSrc $ URecordTy <$> interfaceRecordFields ":"
-  let tyCon = UConDef interfaceName $ toNest [oneAnnBinder]
   let consName = mkInterfaceConsName interfaceName
-  let varTypeName = case oneAnnBinder of
-        Bind v   -> varName v
-        -- This will fail during typechecking.
-        Ignore _ -> mkName "_"
+  let (varTypeName, oneAnnBinder') = case oneAnnBinder of
+        Bind v    -> (varName v, oneAnnBinder)
+        Ignore ty -> let genVar = mkName "gen#" in (genVar, Bind (genVar:>ty))
```

Comment:
Would be nice to add a comment saying that the actual name doesn't matter, but we have to generate something which will never shadow any user names, so we pick this. At least that's what happens IIUC.